src/mon/Monitor.cc

```diff
@@ -3329,40 +3329,63 @@ void Monitor::handle_route(MonOpRequestRef op)
   }
 }
 
+namespace {
+  struct ReentrantLock {
+    const bool initially_locked;
+    Mutex &mutex;
+    ReentrantLock(Mutex& m)
+      : initially_locked(m.is_locked_by_me()),
```

Comment:
so we have following call stack:

Monitor::init() => with lock: bootstrap() => win_standalone_election() => win_election() => Monitor::finish_election() => Monitor::resend_routed_requests()

Elector::expire() => Elector::victory() => **with lock:** Monitor::win_election() => Monitor::finish_election() => Monitor::resend_routed_requests()

Elector::expire() => **with lock:** bootstrap() => win_standalone_election() => win_election() => Monitor::finish_election() => Monitor::resend_routed_requests()

the "with lock:" in bold font are added in this PR.